src/vm/eventpipeconfiguration.cpp

```diff
@@ -404,7 +419,9 @@ void EventPipeConfiguration::DeleteDeferredProviders()
             delete(pProvider);
         }
 
+        SListElem<EventPipeProvider*> *pCurElem = pElem;
```

Comment:
This should not happen unconditionally.  GetDeleteDeferred() is true if the provider was unloaded during tracing.  If it wasn't unloaded, then it can be used in a subsequent session, and the element should not be removed from the list.